substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrGlobalMemory.java

```diff
@@ -141,13 +142,13 @@ private static void releasePromotionBuffer(JfrBuffer buffer) {
     }
 
     @Uninterruptible(reason = "Epoch must not change while in this method.")
-    private static void discardOldest() {
-        // TODO: implement
+    private static boolean shouldDiscard() {
+        // Currently unused as in memory recording is not supported yet.
+        return false;
     }
 
     @Uninterruptible(reason = "Epoch must not change while in this method.")
-    private static boolean shouldDiscard() {
-        // TODO: implement
-        return false;
+    private static void discardOldest() {
+        VMError.unimplemented();
```

Comment:
Looks like `VMError.unimplemented()` is not uninterruptible, so maybe temporarily set callee to be false?